### DISCUSSION TITLE
3.5.0.0-beta1

### DISCUSSION THREAD

**Discussion by Ghost-chu** [QuickShop-Community/QuickShop-Hikari#576] [Announcements]
Today is my birthday! 🎂  And here is birthday update!

**This is a beta build, don't install on live server unless you have full backup.**

## New Features
* DiscordSRV addon (beta), it's a standalone plugin and need you install [DiscordSRV](https://modrinth.com/plugin/discordsrv).
  * Players who linked with their Discord in game now can receive the discord alerts through DM.
    * Including:
      * Anyone sold/bought items to/from your shop. 
      * The shop permission settings changed.
      * The shop transfer to you.
      * The shop has run out of space.
      * The shop has run out of stock.
   * Same as QuickShop-Hikari it self, the messages send to players though Discord also use client display languages.
     * If player offline, QuickShop-Hikari will try to use latest known client language.
* User now able to add new languages by create a folder in `overrides` folder with locale code as name, and same name yaml file inside folder.
  * You can create a locale named `hack` for example, and add a same name language to game client by using Resources Pack, and QuickShop-Hikari will detect new language and loading translation from custom languages.

## Bug Fixes
* Rewrtitten SimpleTextManager, CrowdinOTA and LanguageFilesManager
  * Replaced CrowdinOTA codes to a standalone, bug-free library [CrowdinOTA](https://github.com/Ghost-chu/CrowdinOTA)
    * This fixed incorrect cache handling, QuickShop-Hikari can correct handling the translations that cached in disk, and validating them when next loading to make sure files exists, up-to-date.
    * This also make translation cache files name be more friendly (human readable).
  * Rewritten files loading logic in SimpleTextManager.
    * This fixed incorrect override order and incorrect whole section override.
    * It always loading files by order: built-in failback -> bundled -> crowdin -> user override
    * Fixed strings lost issue if user or crowdin or bundled translations only override part of strings in a section.
    * Fixed translations merge function.
    * Fixed third-party addons translations registering function.

## Developer
* `ShopPlayerGroupSetEvent` now provides player's uuid in event.

## Misc
* Bump adventure-minimessage-platform from 4.1.2 to 4.2.0.

## Screenshots

There is quick preview of new features:

![](https://user-images.githubusercontent.com/30802565/206912863-aecd677f-194a-40bf-8f2c-564f7e3be368.png)
![](https://user-images.githubusercontent.com/30802565/206912864-34a5114a-53c7-41d4-9931-75532c6525b8.png)
![](https://user-images.githubusercontent.com/30802565/206912867-01a5b862-82d3-4988-929d-d6db20f0dd44.png)
![](https://user-images.githubusercontent.com/30802565/206912869-2b0609a8-46f8-4ba0-8a8e-9e2afb77e0a9.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/Ghost-chu/QuickShop-Hikari/releases/tag/3.5.0.0-beta1'>3.5.0.0-beta1</a>.</em>

**Comment by lss233** [comment]
Happy birthday! 🍰

**Comment by YuanYuanOwO** [comment]
Happy Birthday🍰❤️
